features/cellular/framework/AT/AT_CellularNetwork.cpp

```diff
@@ -898,7 +916,14 @@ nsapi_error_t AT_CellularNetwork::detach()
     _at.resp_start();
     _at.resp_stop();
 
-    call_network_cb(NSAPI_STATUS_DISCONNECTED);
+    _at.cmd_start("AT+COPS=2");
```

Comment:
This deregisters from network. Better be done in registration routine not in this detach from PS domain. Set_registration could have true/false param for register/deregister.

Reply:
@AriParkkila Any update required for this PR?

Reply:
@mirelachirica comment must still be handled, I will try to fix that soon.